lib/ansible/modules/source_control/git.py

```diff
@@ -1198,7 +1197,7 @@ def main():
 
     # Deal with submodules
     submodules_updated = False
-    if recursive and not bare:
+    if submodule_init and not bare:
```

Comment:
Maybe
```suggestion
    if (recursive or submodule_init) and not bare:
```
?

Reply:
My only concern with this is its almost the inverse of the same problem that the original developer had with leaving out `submoudle_init` in the first place.

The goal with the changes was first to make sure its backwards compatible with what was here originally (which is equivalent to `submodule_init=yes, recursive=true`) and to ensure it follows the model of what git actually does.

From that perspective, you cant do a `recursive` submodule init without also doing a `submodule_init` with git, hence why if `submodule_init` is not enabled, then `recursive` is ignored (and the comment for `recursive` was updated to reflect that). If we include both here then it does not match the standalone git behavior.

Reply:
Fair enough